File: best-practices/sdk-examples/typescript/streaming-example.ts
import Anthropic from '@anthropic-ai/sdk';
import * as dotenv from 'dotenv';

dotenv.config();

async function main() {
    const anthropic = new Anthropic({
        apiKey: process.env.ANTHROPIC_API_KEY,
    });
    
    console.log('Asking Claude to write a story (streaming)...');
    console.log('-'.repeat(50));
    
    const stream = await anthropic.messages.create({
        model: 'claude-3-5-sonnet-20241022',
        max_tokens: 500,
        messages: [{
            role: 'user',
            content: 'Write a short story about a robot learning to paint'
        }],
        stream: true,
    });
    
    for await (const messageStreamEvent of stream) {
        if (messageStreamEvent.type === 'content_block_delta') {
            process.stdout.write(messageStreamEvent.delta.text);
        }
    }
    
    console.log('\n' + '-'.repeat(50));
    console.log('Stream complete!');
}

main().catch(console.error);

File: best-practices/sdk-examples/python/streaming_example.py
#\!/usr/bin/env python3
"""
Streaming Response Example
Shows how to use Claude's streaming API for real-time responses
"""
import os
from anthropic import Anthropic

def main():
    client = Anthropic(api_key=os.environ.get("ANTHROPIC_API_KEY"))
    
    print("Asking Claude to write a story (streaming)...")
    print("-" * 50)
    
    with client.messages.stream(
        model="claude-3-5-sonnet-20241022",
        max_tokens=500,
        messages=[
            {"role": "user", "content": "Write a short story about a robot learning to paint"}
        ]
    ) as stream:
        for text in stream.text_stream:
            print(text, end='', flush=True)
    
    print("\n" + "-" * 50)
    print("Stream complete\!")

if __name__ == "__main__":
    main()
EOF < /dev/null